### PR TITLE
Remove gang scheduling in RayXShards

### PR DESCRIPTION
**Pull request by yushan111**

### CODE REVIEW

File: pyzoo/zoo/orca/data/ray_xshards.py [hunk 1]
```diff
@@ -190,27 +185,47 @@ def transform_shards_with_actors(self, actors, func,
...
+        list of ray object refs, otherwise return a list of ray objects. Defaults to be False,
+        """
+        assigned_partitions, _ = self.assign_partitions_to_actors(actors)
+        result_refs = []
+        for actor, part_ids in zip(actors, assigned_partitions):
+            assigned_partition_refs = self._get_multiple_partition_refs(part_ids)
+            result_ref = reduce_partitions_func(actor, assigned_partition_refs)
+            result_refs.append(result_ref)
+        if return_refs:
+            return result_refs
```
Comment (yangw1234) [inline pyzoo/zoo/orca/data/ray_xshards.py:208]: why returning `result_refs`, isn't it a list of ray object refs?

File: pyzoo/zoo/orca/data/utils.py [hunk 2]
```diff
@@ -285,8 +297,8 @@ def _convert_list_tuple(data, allow_tuple, allow_list):
 def process_spark_xshards(spark_xshards, num_workers):
     from zoo.orca.data.ray_xshards import RayXShards
     data = spark_xshards
-    if data.num_partitions() != num_workers:
-        data = data.repartition(num_workers)
+    # if data.num_partitions() != num_workers:
```
Comment (yangw1234) [inline pyzoo/zoo/orca/data/utils.py:300]: remove these lines?

File: pyzoo/zoo/orca/data/ray_xshards.py [hunk 3]
```diff
@@ -190,27 +185,47 @@ def transform_shards_with_actors(self, actors, func,
...
+        results = ray.get(result_refs)
+        return results
+
+    def zip_shards_with_actors(self, xshards, actors, reduce_partitions_func, return_refs=False):
```
Comment (yangw1234) [inline pyzoo/zoo/orca/data/ray_xshards.py:212]: `def zip_reduce_shards_with_actors`

**Comment by yangw1234** [comment]
Do we have tests covering the case where input xshard partition number is larger than worker number?

File: pyzoo/zoo/orca/data/utils.py (hunk 2 above)
Comment (yushan111) [reply pyzoo/zoo/orca/data/utils.py:300]: Sure

**Comment by yushan111** [comment]
Jenkins:
http://10.239.47.210:18888/view/ZOO-PR/job/ZOO-PR-Validation/5178/
http://10.239.47.210:18888/job/ZOO-PR-Python-Spark-3.0-py36/593/
http://10.239.47.210:18888/job/ZOO-PR-Validation-Other/5190/

**Comment by yangw1234** [comment]
LGTM

**Comment by yushan111** [comment]
> Do we have tests covering the case where input xshard partition number is larger than worker number?

Yes. We used to force xshard partition number to be equal to the worker number for gang scheduling, and the actual input xshard partition number is larger than worker number in our tests.

File: pyzoo/zoo/orca/data/ray_xshards.py (hunk 3 above)
Comment (yushan111) [reply pyzoo/zoo/orca/data/ray_xshards.py]: Done

**Comment by yushan111** [comment]
I will merge this PR and address further comments in new PR.

**Pull request merged**